src/adaptations/device-layer/include/Weave/DeviceLayer/internal/GenericConnectivityManagerImpl_Thread.ipp

```diff
@@ -62,6 +66,26 @@ void GenericConnectivityManagerImpl_Thread<ImplClass>::_OnPlatformEvent(const We
     {
         UpdateServiceConnectivity();
     }
+
+#if WARM_CONFIG_SUPPORT_THREAD_ROUTING
+    const bool threadRoleChanged = (event->Type == DeviceEventType::kThreadStateChange &&
+                                    event->ThreadStateChange.RoleChanged);
+    if (threadRoleChanged)
+    {
+        otDeviceRole role;
+        ThreadStackMgrImpl().LockThreadStack();
+        role = otThreadGetDeviceRole(ThreadStackMgrImpl().OTInstance());
+        ThreadStackMgrImpl().UnlockThreadStack();
```

Comment:
This class should not call OT directly.  Instead, it should call a method on ThreadStackMgrImpl() which then calls OT.  That way all OT-specific code is kept within the ThreadStackManager object.

I think the right API to call here is ThreadStackMgrImpl().GetThreadDeviceType(). This will tell us if the device has been configured to operate as a router.